pkg/templates/pdb/template.go

```diff
@@ -0,0 +1,114 @@
+package pdb
+
+import (
+	"fmt"
+
+	"golang.stackrox.io/kube-linter/pkg/check"
+	"golang.stackrox.io/kube-linter/pkg/config"
+	"golang.stackrox.io/kube-linter/pkg/diagnostic"
+	"golang.stackrox.io/kube-linter/pkg/lintcontext"
+	"golang.stackrox.io/kube-linter/pkg/objectkinds"
+	"golang.stackrox.io/kube-linter/pkg/templates"
+	"golang.stackrox.io/kube-linter/pkg/templates/pdb/internal/params"
+	v1 "k8s.io/api/apps/v1"
+	pdbV1 "k8s.io/api/policy/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+const (
+	templateKey           = "pod-disruption-budget"
+	maxUnavailableZeroMsg = "MaxUnavailable is set to 0"
+)
+
+func init() {
+	templates.Register(check.Template{
+		HumanName:   "Pod Disruption Budget Configuration Checks",
+		Key:         templateKey,
+		Description: "Flag non-optimal PDB configurations",
+		SupportedObjectKinds: config.ObjectKindsDesc{
+			ObjectKinds: []string{
+				objectkinds.PodDisruptionBudget},
+		},
+		Parameters:             params.ParamDescs,
+		ParseAndValidateParams: params.ParseAndValidate,
+		Instantiate: params.WrapInstantiateFunc(func(p params.Params) (check.Func, error) {
+			return func(lintCtx lintcontext.LintContext, object lintcontext.Object) []diagnostic.Diagnostic {
+
+				var results []diagnostic.Diagnostic
+
+				// Get the PDB provided
+				pdb, ok := object.K8sObject.(*pdbV1.PodDisruptionBudget)
+				if !ok {
+					return nil
+				}
+
+				// If MaxUnavailable is set to 0,
+				// return as no other checks will uniquely apply
+				if pdb.Spec.MaxUnavailable.IntVal == 0 {
+					return []diagnostic.Diagnostic{{
+						Message: maxUnavailableZeroMsg,
+					},
+					}
+				}
+
+				labelSelector, err := metaV1.LabelSelectorAsSelector(&metaV1.LabelSelector{
+					MatchLabels:      pdb.Spec.Selector.MatchLabels,
+					MatchExpressions: pdb.Spec.Selector.MatchExpressions},
+				)
+
+				if err != nil {
+					return []diagnostic.Diagnostic{{
+						Message: fmt.Sprintf("pdb has invalid podSelector: %v", err),
+					}}
+				}
+
+				// Evaluate deployments in the lintContext to see if they have MinAvailable set too low
+				deployments, err := getDeployments(lintCtx, pdb.Namespace, labelSelector)
+				if err != nil {
+					return []diagnostic.Diagnostic{{
+						Message: err.Error(),
+					}}
+				}
+
+				for _, deployment := range deployments {
+					if *deployment.Spec.Replicas == pdb.Spec.MinAvailable.IntVal {
+						results = append(results, diagnostic.Diagnostic{
+							Message: fmt.Sprintf("Deployment %s has replicas equal to the minimum available replicas set by its PDB.", deployment.Name),
+						})
+					}
+				}
+				return results
+
+			}, nil
+		}),
+	})
+}
+
+func getDeployments(lintCtx lintcontext.LintContext, namespace string, labelSelector labels.Selector) ([]v1.Deployment, error) {
+
+	var deploymentList []v1.Deployment
```

Comment:
nit (in conjunction with the comment below on the refactoring of the function):
```suggestion
	objectList := make([]k8sutil.Object, 0, len(lintCtx.Objects()))
```